In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

In [12]:
def total_df(merged, name_str):
  temp = pd.read_csv(f'/content/drive/MyDrive/final_project/{name_str}.csv',index_col=0)
  temp.drop_duplicates(subset='headline',inplace=True)
  return pd.concat([merged,temp])

names = ['samsung','kakao','kb','lgchem','naver','celltrion','hyundai','posco','shinhan','skhynix']
df = pd.DataFrame()
for name in names:
  df = total_df(df, name)

df 

name                                     headline        date  change
0        삼성전자           포스코, 순천시 해룡산단 마그네슘 판재공장 430억 조기 투자  2017-01-02       1
1        삼성전자                   "혁신·변화로 위기 돌파"...대기업 새해 다짐  2017-01-02       1
2        삼성전자                     '노트7' 발화 원인 이르면 이달 중순 발표  2017-01-02       1
3        삼성전자                    경제 위기감 커진 새해…대기업 총수들 “변화”  2017-01-02       1
4        삼성전자                        재계 신년 화두는 ‘재도약·신뢰 회복’  2017-01-02       1
...       ...                                          ...         ...     ...
24543  SK하이닉스              '삼천피' 못지킨 2021 증시…그래도 '천스닥'은 꿋꿋  2021-12-30       1
24544  SK하이닉스  "코로나 봉쇄로 삼성 시안공장 생산 줄여…메모리 공급사 주가에는 긍정적 영향"  2021-12-30       1
24545  SK하이닉스                         SK하이닉스, 인텔 낸드 1단계 인수  2021-12-30       1
24546  SK하이닉스                  삼성전자만 11조원…연기금, 올해 24조원 팔았다  2021-12-30       1
24547  SK하이닉스                        SK하이닉스, 특별성과급 300% 쏜다  2021-12-30       1

[212244 rows x 4 columns]

In [15]:
df.reset_index(inplace=True)
del df['index']
df

name                                     headline        date  change
0         삼성전자           포스코, 순천시 해룡산단 마그네슘 판재공장 430억 조기 투자  2017-01-02       1
1         삼성전자                   "혁신·변화로 위기 돌파"...대기업 새해 다짐  2017-01-02       1
2         삼성전자                     '노트7' 발화 원인 이르면 이달 중순 발표  2017-01-02       1
3         삼성전자                    경제 위기감 커진 새해…대기업 총수들 “변화”  2017-01-02       1
4         삼성전자                        재계 신년 화두는 ‘재도약·신뢰 회복’  2017-01-02       1
...        ...                                          ...         ...     ...
212239  SK하이닉스              '삼천피' 못지킨 2021 증시…그래도 '천스닥'은 꿋꿋  2021-12-30       1
212240  SK하이닉스  "코로나 봉쇄로 삼성 시안공장 생산 줄여…메모리 공급사 주가에는 긍정적 영향"  2021-12-30       1
212241  SK하이닉스                         SK하이닉스, 인텔 낸드 1단계 인수  2021-12-30       1
212242  SK하이닉스                  삼성전자만 11조원…연기금, 올해 24조원 팔았다  2021-12-30       1
212243  SK하이닉스                        SK하이닉스, 특별성과급 300% 쏜다  2021-12-30       1

[212244 rows x 4 columns]

### 전체 데이터셋 구성
- 2016.01.01-2021.12.31 (주식 시장 개장일기준 6년 데이터 최대 20개씩 크롤링)
- 2016 - 2020은 train, 2021은 test로 사용 예정


- title 기준 중복 제거 전 215144개
- title 기준 중복 제거 후 212244개
  - 약 3000개의 중복 headline 제거

In [18]:
df['date'] = pd.to_datetime(df['date'])
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 212244 entries, 0 to 212243
Data columns (total 4 columns):
 #   Column    Non-Null Count   Dtype         
---  ------    --------------   -----         
 0   name      212244 non-null  object        
 1   headline  212244 non-null  object        
 2   date      212244 non-null  datetime64[ns]
 3   change    212244 non-null  int64         
dtypes: datetime64[ns](1), int64(1), object(2)
memory usage: 6.5+ MB


In [19]:
df.to_csv('/content/drive/MyDrive/final_project/crawling_df.csv',encoding='utf-8')